src/hotspot/cpu/x86/assembler_x86.cpp

```diff
@@ -12955,6 +12925,29 @@ void Assembler::eevex_prefix_ndd(Address adr, int ndd_enc, int xreg_enc, VexSimd
   vex_prefix(adr, ndd_enc, xreg_enc, pre, opc, attributes, /* nds_is_ndd */ true, no_flags);
 }
 
+void Assembler::emit_eevex_or_demote(Register dst, Address src1, Register src2, VexSimdPrefix pre, VexOpcode opc,
+                                     int size, int opcode_byte, bool no_flags, bool is_map1, bool is_commutative) {
+  if (is_commutative && is_demotable(no_flags, dst->encoding(), src2->encoding())) {
+    if (size == EVEX_64bit) {
+      emit_prefix_and_int8(get_prefixq(src1, dst, is_map1), opcode_byte + 2);
```

Comment:
It will be good to write a comment on top of opcode_byte adjustment on account of opcode mismatch b/w NDD and equivalent demotable variant.

```
EVEX.LLZ.NP.MAP4.SCALABLE 21 /r      AND {NF} {ND=1} rv, rv/mv, rv
```

`REX.W + 23 /r      AND r64, r/m64 | RM | Valid | N.E. | r64 AND r/m64
`